File: projects/03/grade_calc.rb
#! /usr/bin/ruby
# grade_calc.rb translates an integer represented grade to a letter grade
# Author: Josiah Ryan
# Date: 02/24/21
##############################################################################

# Input:  a integer grade
# Precondition: The the grade should be between 0 and 100
# Output: the letter grade corresponding to the int grade given

def gradeCalc(grade)
  case grade / 10 # switch-case
  when 10, 9
    "A"
  when 8
    "B"
  when 7
    "C"
  when 6
    "D"
  else
    "F"
  end
end

if __FILE__ == $0
  print "Enter a numeric grade between 0-100: "
  grade = gets            #gets input
  print "Your letter grade is: "
  puts gradeCalc(grade.to_i) # .to_i converts string to int
end

File: projects/03/clojure/src/grade_calc.clj
;;;; grade_calc.clj is a driver for function gradeCalc.
;;;;
;;;; Input:  A grade out of 100
;;;; Output: The letter grade cooresponding to that number grade
;;;;
;;;; Usage: clojure -m grade_calc
;;;;
;;;; Author: Josiah Ryan
;;;; Date: 02/24/21
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns grade_calc)			; name of the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; gradeCalc() returns the letter grade corresponding to the given integer grade.
;;; Receive: grade, an int.
;;; Precondition: grade is between 0 and 100
;;; Return: the letter grae corresponding to the int grade given
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn gradeCalc [grade]
  ;(cond ((== (quot grade 10) 10) 'A'))
  (cond (== (quot grade 10) 10) 'A   ; quot is division
        (== (quot grade 10) 9)  'A
	(== (quot grade 10) 8)  'B
	(== (quot grade 10) 7)  'C
	(== (quot grade 10) 6)  'D
	:else         'F
  )
)

(defn parse-int [s]
   (Integer. (re-find  #"\d+" s )))

(defn -main []
 (print "\nEnter your numeric grade: ") (flush)
 (let
   [ grade (read-line) ]		; reads grade (a string)
   (println "Your letter grade is: " (gradeCalc (parse-int grade)) "\n") 	;display the letter grade
 )
)